### PR TITLE
Update branch in gitmodules for fleet prov library

### PR DESCRIPTION
**Pull request by nateglims**
*Description of changes:* Adding the branch specifier to .gitmodules for the fleet provisioning library so the CI system (and others) can accurately update the submodule to the intended branch rather than wait for submodule reference to be updated by the library authors.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -42,6 +42,7 @@
 	url = https://github.com/eclipse/mosquitto.git
 [submodule "libraries/aws/fleet-provisioning-for-aws-iot-embedded-sdk"]
 	path = libraries/aws/fleet-provisioning-for-aws-iot-embedded-sdk
+    branch = main
```
Comment (aggarw13) [inline .gitmodules:45]:
```suggestion
     branch = main
```

Comment (archigup) [inline .gitmodules:45]:
```suggestion
	branch = main
```

File: .gitmodules [hunk 2]
```diff
@@ -42,6 +42,7 @@
...
 	url = https://github.com/aws/Fleet-Provisioning-for-AWS-IoT-embedded-sdk
```
Comment (aggarw13) [inline .gitmodules:46]:
```suggestion
[submodule "libraries/aws/fleet-provisioning-for-aws-iot-embedded-sdk"]
	path = libraries/aws/fleet-provisioning-for-aws-iot-embedded-sdk
        branch = main
	url = https://github.com/aws/Fleet-Provisioning-for-AWS-IoT-embedded-sdk
```

**Review by archigup — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Comment by nateglims** [comment]
Re-running checks because the mbed forums were temporarily down.

**Pull request merged**